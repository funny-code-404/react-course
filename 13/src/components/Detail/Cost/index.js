import PropTypes from "prop-types";

const Cost = ({ data }) => (
	data && 
	<div>
		<p>Costs:</p>
		<ul>
			{Object.keys(data).map((e, i)=> (
				<li key={i} >{e}<span>{data[e]}</span></li>
			))}
		</ul>
	</div>
)

Cost.propTypes = {
	data: PropTypes.objectOf(PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
	])).isRequired,
}


export default Cost;