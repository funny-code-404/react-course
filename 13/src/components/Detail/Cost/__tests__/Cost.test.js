import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Cost from "..";

configure({ adapter: new Adapter() });

describe('DETAIL COST', () => {
    const props = {
        data: {
            Wood: 1,
        }
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Cost {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('Component is return 1 li content', () => {
        expect(wrapper.find('li')).toHaveLength(1);
    });
    it('Component is return 1 span content', () => {
        expect(wrapper.find('span')).toHaveLength(1);
    });
    it('Component should render correct data content', () => {
        expect(wrapper.find('span').props().children).toEqual(props.data.Wood);
    });
    it('Component should render correct after props changing', () => {
        const newdata = 'NEW';
        wrapper.setProps({ data: { Wood: newdata }})
        expect(wrapper.find('span').props().children).toEqual(newdata);
    });
});

// npm run test src/components/Detail/Cost