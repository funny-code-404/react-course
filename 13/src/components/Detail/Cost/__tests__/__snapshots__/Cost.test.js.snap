// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cost to Match Snapshot 1`] = `
<div>
  <p>
    Costs:
  </p>
  <ul>
    <li>
      Wood - 
      <span>
        1
      </span>
    </li>
  </ul>
</div>
`;

exports[`DETAIL COST to Match Snapshot 1`] = `
<div>
  <p>
    Costs:
  </p>
  <ul>
    <li>
      Wood - 
      <span>
        1
      </span>
    </li>
  </ul>
</div>
`;
